### PR TITLE
Laravel 5 Compatibiity

### PR DESCRIPTION
**Pull request by impleri**
Laravel 5 has a few breaking changes (namely RemindableInterface is replaced by Authenticable interface) and a new interface for the main Application. I've updated tests and code to match these two changes, as well as composer dependencies on just 5.x. I also added the last few tests needed to get full code coverage (though that may be more a byproduct of my futzing with everything else).

### CODE REVIEW

**Comment by impleri** [comment]
Touching #448 .

**Comment by JeramyMywork** [comment]
This is working great for me :+1:  Can we get this pulled into the main git repository please?
At least into a laravel-5 branch

**Comment by andrewelkins** [comment]
Yeah I'll get it merged in to the L5 branch shortly. Need to do that from the command line, so it'll have to wait till tonight.

**Comment by od3n** [comment]
+1

**Comment by xtrasmal** [comment]
hello party animals. It has been a while... how is it hanging?

**Comment by Ye-Ting** [comment]
+1

**Comment by zh-daniel-huang** [comment]
ping @andrewelkins

**Comment by BlueManLine** [comment]
and...

File: src/Confide/CacheLoginThrottleService.php [hunk 1]
```diff
@@ -22,9 +24,9 @@ class CacheLoginThrottleService implements LoginThrottleServiceInterface
      *
      * @param \Illuminate\Foundation\Application $app Laravel application object.
```
Comment (GrahamCampbell) [inline src/Confide/CacheLoginThrottleService.php]: php doc needs updating

File: src/Confide/Confide.php [hunk 2]
```diff
@@ -52,12 +54,12 @@ public function __construct(
         RepositoryInterface       $repo,
         PasswordServiceInterface  $passService,
         LoginThrottleServiceInterface $loginThrottler,
-        $app = null
+        Application $app
```
Comment (GrahamCampbell) [inline src/Confide/Confide.php]: phpdoc needs updating in all cases

File: composer.json [hunk 3]
```diff
@@ -15,14 +15,13 @@
     ],
     "require": {
         "php": ">=5.4.0",
-        "illuminate/support": "~4.2|~5.0"
+        "illuminate/support": "~5.0"
```
Comment (GrahamCampbell) [inline composer.json]: You should really add contracts as a dependency, even though it's currently required by support.

Comment (GrahamCampbell) [inline composer.json]: `~5.0` is wrong. YOu don't know if this will still work on L5.2/5.6 etc,

Comment (GrahamCampbell) [inline composer.json]: Better to make it `5.1.*`. No point in supporting 5.0.x as it's no longer supported by laravel anymore.

Comment (impleri) [inline composer.json]: I can update this, but note that this PR is 2 months old, long before 5.1 was released. Also, isn't Laravel using semver? Considering how long it's taken for a code review here, making it 5.1.x will be great until 5.2 is released.

Also, are you sure 5.0.x isn't supported anymore: http://laravel.com/docs/5.1/releases ?
>  For general releases, bug fixes are provided for 6 months and security fixes are provided for 1 year.

**Comment by dcrawkstar** [comment]
Any word on getting to that command line? Would enjoy seeing this support for L5

**Comment by impleri** [comment]
Authenticable was a typo.

Documentation updated to refer to Contracts namespace.

**Comment by impleri** [comment]
Leaving version as ~5.0, though, because that is pretty standard version handling in semver. Until Laravel documents its versioning schema, I default to semver.

**Pull request closed without merging**